src/libraries/System.Private.Xml/src/System/Xml/Cache/XPathDocumentIterator.cs

```diff
@@ -164,8 +166,9 @@ public override bool MoveNext()
     /// </summary>
     internal class XPathDocumentElementDescendantIterator : XPathDocumentBaseIterator
     {
-        private readonly XPathDocumentNavigator _end;
-        private readonly string _localName, _namespaceUri;
+        private readonly XPathDocumentNavigator? _end;
+        private readonly string? _localName;
+        private readonly string _namespaceUri;
```

Comment:
Line 181 (can't put comment there): I'd expect `_localName = root.NameTable.Get(name);` to never return null in normal situation (presumably all names are already atomized) but fine to leave as nullable since there is no check

Reply:
NameTable.Get can return null if the value is not in `_entries`. I could change `_localName` to not-nullable and add a `Debug.Assert` after the `Get` call if you think that's more appropriate.

Reply:
I meant that this class is constructed from already existing document which presumably already has all entries in the name table. I don't feel strongly on changing this since it's an implementation detail and does not affect public APIs

Reply:
Keep in mind that method `XPathDocumentNavigator.IsElementMatch` which receives this field has a null check and it is only being called from this class.